### PR TITLE
Updated YANG model for IETF 115

### LINKED ISSUE #107: Optical protection switching requires additional parameters
In case optical protection switching is used , the transceiver output is splitted onto 2 paths. From our model point of view, a single OTSi is referenced at the transceiver, which means that the two paths share the same OTSi.
First this cause an issue both in the case of regeneration on one of the path. Secondly, the case where transceivers are outside of the network (then we don't know where the loop terminates) also causes an issue because it forms a loop.
I attach slides to explain the issue.
[Issue for optical protection switching representation.pptx](https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/files/8398878/Issue.for.optical.protection.switching.representation.pptx)
.

### PR DESCRIPTION
**Pull request by italobusi**
Updated optical impairments YANG model:
- OPS capability reporting: fix #107
- fix augmentation when statements

Pending merge of PR https://github.com/ietf-ccamp-wg/ietf-ccamp-layer0-types-ext/pull/60

Co-Authored-By: sergio belotti <sergio.belotti@nokia.com>

### CODE REVIEW

File: ietf-optical-impairment-topology.yang [hunk 1]
```diff
@@ -1264,17 +1287,122 @@ module ietf-optical-impairment-topology {
...
+         + "'optical-imp-topo:otsi-protection')" {
+        description
+          "This list applies only to 'to' LTPs that support 
+          individual OTSi(G) protection.";
+      }
+      key "ltp-ref";
+      config false;
+      description
+        "The restricted list of the potential secondary LTPs that 
+        can be selected when the 'from' LTP of this connectivity
```
Comment (EstherLerouzic) [inline ietf-optical-impairment-topology.yang:1369]: shouldn't this be 'to' instead of 'from' ?

File: ietf-optical-impairment-topology.yang [hunk 2]
```diff
@@ -1264,17 +1287,122 @@ module ietf-optical-impairment-topology {
...
+        If this list is empty, all the other LTPs that can reach 
+        the 'to' LTP of this connectivity matrix entry can be 
+        selected as secondary LTPs.";
+      leaf ltp-ref {
+        type leafref {
+          path "../../../../../../../nt:termination-point/nt:tp-id";
+        }
+        description
+          "The reference to the potential secondary LTP that can be 
+          selected when the 'from' LTP of this connectivity matrix
```
Comment (EstherLerouzic) [inline ietf-optical-impairment-topology.yang:1381]: shouldn't this be "The reference to the potential secondary LTP that can be 
          selected when the **'to'** LTP of this connectivity matrix 
          entry is selected as a working LTP";

File: ietf-optical-impairment-topology.yang [hunk 3]
```diff
@@ -1264,17 +1287,122 @@ module ietf-optical-impairment-topology {
...
+      }
+      leaf roadm-path-impairments {
+        type leafref {
+          path "../../../../../../../tet:te/tet:te-node-attributes/"
+             + "roadm-path-impairments/roadm-path-impairments-id";
+        }
+        config false;
+        description
+          "Pointer to ROADM optical impairments of the ROADM path 
+          between the secondary LTP and the 'to' LTP of this
```
Comment (EstherLerouzic) [inline ietf-optical-impairment-topology.yang:1392]: same issue

File: ietf-optical-impairment-topology.yang [hunk 4]
```diff
@@ -1264,17 +1287,122 @@ module ietf-optical-impairment-topology {
...
+      connectivity matrix entry.";
+    list additional-ltp {
+      when "derived-from-or-self(../../../../../../"
+         + "nt:termination-point[nt:tp-id=current()/../tet:tp-ref]/"
+         + "optical-imp-topo:protection-type,"
+         + "'optical-imp-topo:otsi-protection')" {
+        description
+          "This list applies only to 'from' LTPs that support 
+          individual OTSi(G) protection.";
+      }
```
Comment (sergiobelotti) [inline ietf-optical-impairment-topology.yang:1310]: this list in the "from" LTPs applies only when the "to" LTP support OPS protection

File: ietf-optical-impairment-topology.yang [hunk 5]
```diff
@@ -1264,17 +1287,122 @@ module ietf-optical-impairment-topology {
...
+      connectivity matrix entry.";
+    list additional-ltp {
+      when "derived-from-or-self(../../../../../../"
+         + "nt:termination-point[nt:tp-id=current()/../tet:tp-ref]/"
+         + "optical-imp-topo:protection-type,"
+         + "'optical-imp-topo:otsi-protection')" {
+        description
+          "This list applies only to 'to' LTPs that support 
+          individual OTSi(G) protection.";
+      }
```
Comment (sergiobelotti) [inline ietf-optical-impairment-topology.yang:1364]: this list in the "to" LTPs applies only when the "from" LTP supports OPS protection

File: ietf-optical-impairment-topology.yang (hunk 1 above)
Comment (italobusi) [reply ietf-optical-impairment-topology.yang]: Fixed in https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/pull/116/commits/91193f0458ddb0b0bf3c4ec40e73356df546d39e

File: ietf-optical-impairment-topology.yang (hunk 2 above)
Comment (italobusi) [reply ietf-optical-impairment-topology.yang]: Fixed in https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/pull/116/commits/91193f0458ddb0b0bf3c4ec40e73356df546d39e

File: ietf-optical-impairment-topology.yang (hunk 3 above)
Comment (italobusi) [reply ietf-optical-impairment-topology.yang]: Fixed in https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/pull/116/commits/91193f0458ddb0b0bf3c4ec40e73356df546d39e

File: ietf-optical-impairment-topology.yang (hunk 4 above)
Comment (italobusi) [reply ietf-optical-impairment-topology.yang]: Fixed in https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/pull/116/commits/91193f0458ddb0b0bf3c4ec40e73356df546d39e

File: ietf-optical-impairment-topology.yang (hunk 5 above)
Comment (italobusi) [reply ietf-optical-impairment-topology.yang]: Fixed in https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/pull/116/commits/91193f0458ddb0b0bf3c4ec40e73356df546d39e

**Review by sergiobelotti — APPROVED** [review]

**Review by aguoietf — APPROVED** [review]

**Pull request merged**